int8 exists
float32 x
float32 y
float32 radius
float32 goodness_of_fit